examples/caa/caa.ipynb

```diff
@@ -0,0 +1,590 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Contrastive Activation Addition\n",
+    "\n",
+    "This notebook aims to reproduce the workflow defined in [Contrastive Activation Addition](https://arxiv.org/abs/2312.06681) for extracting steering vectors from input. The official codebase can be found [here](https://github.com/nrimsky/CAA)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To be consistent with CAA, we run on Llama-2 chat models of sizes 7b and 13b. These can be downloaded through Huggingface Transformers but require you to have first applied for access. \n",
+    "\n",
+    "- If you have downloaded the models to the standard location, you can skip the next cell.\n",
+    "- If you have downloaded the models to a custom path, you can set TRANSFORMERS_CACHE to point to your download path. \n",
+    "- If you have not downloaded the models, you will need to apply for access [here](https://ai.meta.com/resources/models-and-libraries/llama-downloads/). Once you have gotten access, you can set your HF_TOKEN to download the models. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# %env HF_TOKEN = <YOUR_HF_TOKEN>\n",
+    "# %env TRANSFORMERS_CACHE = <YOUR_CACHE_DIR>"
```

Comment:
nit: I feel like the `TRANSFORMERS_CACHE` stuff is too much detail for a notebook demo. If a users wants to modify huggingface they can figure out all the options available themselves.

IMO, it might be a nicer experience to use `notebook_login()` to log the user into huggingface in a notebook rather than dealing with setting env vars directly. `notebook_login()` will direct the user to get a token on their own.

Reply:
Great idea, I had no idea `notebook_login()` existed